pandas/core/reshape.py

```diff
@@ -940,24 +942,25 @@ def melt_stub(df, stub, i, j):
         varnames = get_var_names(df, "^" + stub)
         newdf = melt(df, id_vars=i, value_vars=varnames, value_name=stub,
                      var_name=j)
-        newdf_j = newdf[j].str.replace(stub, "")
-        try:
-            newdf_j = newdf_j.astype(int)
-        except ValueError:
-            pass
-        newdf[j] = newdf_j
-        return newdf
+        newdf[j] = Categorical(newdf[j])
+        newdf[j] = newdf[j].str.replace(stub, "")
+        return newdf.set_index([i, j])
 
     id_vars = get_var_names(df, "^(?!%s)" % "|".join(stubnames))
     if i not in id_vars:
         id_vars += [i]
 
-    newdf = melt_stub(df, stubnames[0], id_vars, j)
+    mstubs = [melt_stub(df, stub, i, j) for stub in stubnames]
+    mstubs = mstubs[0].join(mstubs[1:], how='outer')
+    new = df[id_vars].set_index(i).join(mstubs)
+
+    try:
+        new.index.set_levels(new.index.levels[-1].astype(int), level=-1,
```

Comment:
This is just the same int conversion attempt done in the original code [here](https://github.com/pandas-dev/pandas/blob/4814823903b862c411caf527271e384df0d0d7e7/pandas/core/reshape.py#L944), since this "time" column may contain strings not necessarily can be converted to integers.